### DISCUSSION TITLE
How do I set CSS style for specific q-field under another component?

### DISCUSSION THREAD

**Discussion by AdagioB** [quasarframework/quasar#13320] [General - Components / Directives / etc] [answered]
Codepen: https://codepen.io/Adagio_B/pen/rNJaYgj

Hi

In the above codepen, I am trying to set a special style for q-field__append, but only for input fields with a specific id (in this case 'event'), while leave all others untouched

![image](https://user-images.githubusercontent.com/61786026/166415985-71df3457-9804-4848-a1fe-d2a4426f23c7.png)

As can bee seen, I tried to use #event.q-field__append, but the id is not sent to the sub q-field elements. If I just use .q-field__append it will set that style to all q-field elements with append

Alternative is there a way to have a set of style that would only work on that specific vue component?

I have tried to search for a way to find the q-field in JavaScript and add my class to it, but I wasn't able to find this option. From code it doesn't seem like I have access to this specific q-field

**Comment by pdanpdan** [comment, marked as answer]
Don't use `id` for multiple reasons:
- it can be only one
- on components it might do unexpected things (especially on QField/QInput/QSelect it does nothing)

Use a CSS `class`

https://codepen.io/pdanpdan/pen/gOvbKom?editors=1100

**Reply by AdagioB** [reply]
Thank you so much, helped me out a lot here and I learned a bit more
